WinBioNET/WinBio.cs

```diff
@@ -102,18 +103,58 @@ public static WinBioStorageSchema[] EnumDatabases(WinBioBiometricType factor)
             WinBioSessionHandle sessionHandle,
             WinBioBirPurpose purpose,
             WinBioBirDataFlags flags,
-            out int sampleSize,
-            out WinBioRejectDetail rejectDetail)
+            out WinBioRejectDetail rejectDetail,
+            out Size imageSize,
+            out IntPtr firstPixelPointer)
         {
             int unitId;
-            IntPtr pointer;
-            var code = CaptureSample(sessionHandle, purpose, flags, out unitId, out pointer, out sampleSize, out rejectDetail);
+            int sampleSize;
+            IntPtr samplePointer;
+            var code = CaptureSample(sessionHandle, purpose, flags, out unitId, out samplePointer, out sampleSize, out rejectDetail);
             WinBioException.ThrowOnError(code, "WinBioCaptureSample failed");
-            //TODO: parse WINBIO_BIR structure at pointer
-            Free(pointer);
+            WinBioBir sample = (WinBioBir)Marshal.PtrToStructure(samplePointer, typeof(WinBioBir));
+            IntPtr birHeaderPointer = samplePointer + sample.HeaderBlock.Offset;
+            IntPtr ansiHeaderPointer = samplePointer + sample.StandardDataBlock.Offset;
+            IntPtr ansiRecordPointer = ansiHeaderPointer + Marshal.SizeOf(typeof(WinBioBdbAnsi381Header));
+
+            WinBioBdbAnsi381Record ansiRecord = (WinBioBdbAnsi381Record)Marshal.PtrToStructure(
+                ansiRecordPointer, typeof(WinBioBdbAnsi381Record));
+
+            imageSize = new Size(ansiRecord.HorizontalLineLength, ansiRecord.VerticalLineLength);
+            firstPixelPointer = ansiRecordPointer + Marshal.SizeOf(typeof(WinBioBdbAnsi381Record));
```

Comment:
This is problematic, because it will point to memory which is being released before this method returns, see below.

Reply:
You are right. I tried to split the function into two because it was getting long but haven't noticed that.